apiserver/observer/request_notifier.go

```diff
@@ -71,6 +93,9 @@ func (n *RequestObserver) Join(req *http.Request, connectionID uint64) {
 
 // Leave implements Observer.
 func (n *RequestObserver) Leave() {
+	if n.state.agent {
+		n.connLogger.Infof("agent disconnected: %s for %s", n.state.tag, n.state.model)
```

Comment:
as a side note, I've seen some stuff that was using truncated model UUIds, and I wonder if we might want to do something standard about how to represent models that can be more human readable than a 36 character UUID. (Arguably model Name would be best, but truncated 10char UUID might be something more humane, though we'd want to do it consistently)